src/chttpd_auth_cache.erl

```diff
@@ -95,7 +95,8 @@ get_from_cache(UserName) ->
 %% gen_server callbacks
 
 init([]) ->
-    {ok, #state{changes_pid = spawn_changes(0)}}.
+    self() ! {start_listener, 0},
```

Comment:
Thanks for detailed explanation! 
I'm agree with you that all cases that involves dependency on time shouldn't be used. The `#2` sounds good as it will require test suite (tests are good in any case), but I don't have any solid expertise here to judge.